src/App.js

```diff
@@ -1,83 +1,100 @@
-import React from "react";
-import "./App.css";
+import React from 'react';
+import { useEffect, useState } from 'react';
+import './App.css';
+import Weather from './components/Weather';
+import Forecast from './components/Forecast';
+import { cities } from './utils/cities';
 
 const App = () => {
+  const [weather, setWeather] = useState(null);
+  const [forecast, setForcast] = useState(null);
+  const [city, setCity] = useState('Prague');
+  const api_id = process.env.REACT_APP_MY_API_ID;
+
+  const fetchWeather = () => {
+    fetch(
+      `https://api.openweathermap.org/data/2.5/weather?q=${city}&units=metric&appid=${api_id}`,
+    )
+      .then((response) => response.json())
+      .then((data) => {
+        setWeather(data);
+        console.log({ weather });
+      });
+  };
+
+  const fetchForecast = () => {
+    fetch(
+      `https://api.openweathermap.org/data/2.5/forecast?q=${city}&units=metric&appid=${api_id}`,
+    )
+      .then((response) => response.json())
+      .then((data) => {
+        setForcast(data.list.filter((element, index) => index % 8 === 0));
+        console.log({ forecast });
+      });
+  };
+
+  useEffect(() => {
+    fetchWeather();
+    fetchForecast();
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [city]);
```

Comment:
Tohle je dobrý, ale je potřeba si uvědomit že tyhle 2 operace běží paralelně. Tj kdyby např fetchWeather trval 3s (může se někde seknout), a fetchForecast 500ms, tak se ti UI updatuje "po kouskách". Občas to chceš, ale většinou načítání dat poslouchá tenhle pattern:
- nastav loading stav
- načti z API všechna data která potřebuješ
- nastav potřebný stav a vypni loading

Plus ještě chceš ošetřit případ, kdy někdo rychle přepíná města, rychlejc než se stačí načíst. Chceš ty staré výsledky co dojdou později než nové zahodit;

tj v tomhle případě:
```
useEffect(() => {
   let cancelled = false;

   setWeather(null);
   setForecast(null);

   Promise.all([
      fetch(`https://api.openweathermap.org/data/2.5/weather?appid=${API_KEY}&q=${city}&units=metric`).then((response) => response.json()),
fetch(`https://api.openweathermap.org/data/2.5/forecast?appid=${API_KEY}&q=${city}&units=metric`)
    .then((response) => response.json()),
   ]).then(([weather, forecast]) => {
        if(!cancelled){
            setWeather(weather);
            setForecast(forecast.filter((element, index) => index % 8 === 0));
        }
   });

   return () => {
      cancelled = true;
   }
}, [city])
```
Ale to ti říkám jen pro informaci a abys viděla, že jakékoli programování s nějakými asynchroními věcmi začíná být o dost složitější :)